apiserver/facades/client/client/status.go

```diff
@@ -543,24 +566,79 @@ func fetchRemoteApplications(st Backend) (map[string]*state.RemoteApplication, e
 	return appMap, nil
 }
 
+// fetchOfferConnections returns a map from relation id to offer connection.
+func fetchOffers(st Backend) (map[string]offerStatus, error) {
+	offersMap := make(map[string]offerStatus)
+	offers, err := st.AllApplicationOffers()
+	if err != nil {
+		return nil, err
+	}
+	model, err := st.Model()
+	if err != nil {
+		return nil, err
+	}
+	for _, offer := range offers {
+		offersMap[offer.OfferName] = offerStatus{
+			ApplicationOffer: crossmodel.ApplicationOffer{
+				OfferName:       offer.OfferName,
+				ApplicationName: offer.ApplicationName,
+				Endpoints:       offer.Endpoints,
+			},
+			ApplicationURL: crossmodel.MakeURL(model.Owner().Name(), model.Name(), offer.OfferName, ""),
+		}
+	}
+	return offersMap, nil
+}
+
+// fetchOfferConnections returns a map from relation id to offer connection.
+func fetchOfferConnections(st Backend) (map[int]*state.OfferConnection, error) {
+	connMap := make(map[int]*state.OfferConnection)
+	conns, err := st.AllOfferConnections()
+	if err != nil {
+		return nil, err
+	}
+	for _, c := range conns {
+		connMap[c.RelationId()] = c
+	}
+	return connMap, nil
+}
+
 // fetchRelations returns a map of all relations keyed by application name.
 //
 // This structure is useful for processApplicationRelations() which needs
 // to have the relations for each application. Reading them once here
 // avoids the repeated DB hits to retrieve the relations for each
 // application that used to happen in processApplicationRelations().
-func fetchRelations(st Backend) (map[string][]*state.Relation, error) {
+func fetchRelations(st Backend) (map[string][]*state.Relation, map[int]*state.Relation, error) {
 	relations, err := st.AllRelations()
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 	out := make(map[string][]*state.Relation)
+	outById := make(map[int]*state.Relation)
 	for _, relation := range relations {
+		outById[relation.Id()] = relation
+		// If either end of the relation is a remote application
+		// on the offering side, exclude it here.
+		isRemote := false
+		for _, ep := range relation.Endpoints() {
+			if app, err := st.RemoteApplication(ep.ApplicationName); err == nil {
+				if app.IsConsumerProxy() {
```

Comment:
The idea of calling it a ConsumerProxy is because it directly represents a proxy to an app on the consuming side. But IsOfferingSide() works too. Maybe as a drive by next time? Yes, an app can relate to both something on the offering side (eg a subordinate or other collaborating app) and something remote (typically via different endpoints).